ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpConnection.java

```diff
@@ -0,0 +1,23 @@
+package io.ripc.protocol.tcp;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * An abstraction for a TCP connection.
+ *
+ * @param <R> The type of objects read from this connection.
+ * @param <W> The type of objects written to this connection.
+ */
+public interface TcpConnection<R, W> extends Publisher<R> {
```

Comment:
Think about the channel from Netty then or from Java NIO, there is absolutely no need to replicate write for every sort of IO, it's not a big deal but its an API question. It would be like saying we have different type of Subscriber in Reactive Stream for Synchronous and Asynchronous ones. UDP is also handled through Channel. So maybe Channel is a better name overall if we want a common denominator.
At this module level we are API centric not Application centric anyway IMO, feel free to name it RxTcpConnection in rxjava tho if you bring specific value to this.

`You mean file streaming over TCP, HTTP?
No, I am thinking about reading and writing a File. E.g. Chronicle or other File Memory Mapped IPC used in high throughput apps.